osu.Game/Screens/Menu/MainMenu.cs

```diff
@@ -226,15 +236,17 @@ public override void OnSuspending(IScreen next)
 
             buttons.State = ButtonSystemState.EnteringMode;
 
+            songTicker.Hide();
             this.FadeOut(FADE_OUT_DURATION, Easing.InSine);
-            this.MoveTo(new Vector2(-800, 0), FADE_OUT_DURATION, Easing.InSine);
+            buttonsContainer.MoveTo(new Vector2(-800, 0), FADE_OUT_DURATION, Easing.InSine);
 
             sideFlashes.FadeOut(64, Easing.OutQuint);
         }
 
         public override void OnResuming(IScreen last)
         {
             base.OnResuming(last);
+            songTicker.Hide();
```

Comment:
this shouldn't be necessary

Reply:
song ticker can be triggered even in song select screen. So if you are in there, select another beatmap and go back to main menu - you will see the song ticker. That's why I'm hiding it here

Reply:
it shouldn't be processing changes when not at the menu.

Reply:
and how does it even become visible again then, after returning from song select?

Reply:
Good question. Exiting song select without that line:
![nice](https://user-images.githubusercontent.com/22874522/72206179-d5d06100-349b-11ea-96a7-4bf3e1f6616c.gif)

Reply:
As long as you understand it's a question you need to answer. The `Hide` as it stands is not correct.

Reply:
Got it, investigating

Reply:
Simplest (but not sure if correct) solution: I just added `AllowUpdates` flag to song ticker.

Reply:
You're probably looking for `mainMenu.IsCurrentScreen()`, or even more simpler, checking if the ticker `IsPresent`.

Reply:
`IsPresent` will not work because even on the main screen we are triggering changes while fully invisible

Reply:
Right, in that case, pass the menu and check `IsCurrentScreen()`.

Reply:
I personally don't think passing a screen for the component is a good idea. What if (for example) it will be implemented somewhere else? There may be other conditions.

Reply:
That'll be added later, currently it is used once at the main menu and it should not update when it's out of the screen, no need to make it that complex in my opinion.